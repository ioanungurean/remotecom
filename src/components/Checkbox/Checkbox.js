import { StyledCheckbox, StyledInput, StyledControl, StyledLabel } from './Checkbox.styled';

const Checkbox = ({ id, label, onChange }) => {
  const handleChange = () => {
    onChange(id);
  };

  return (
    <StyledCheckbox htmlFor={id}>
      <StyledInput id={id} type="checkbox" onChange={handleChange} />
      <StyledControl />
      <StyledLabel>{label}</StyledLabel>
    </StyledCheckbox>
  );
};

export default Checkbox;
