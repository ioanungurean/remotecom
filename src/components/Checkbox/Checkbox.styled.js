import styled, { css } from 'styled-components';

export const StyledControl = styled.span`
  position: relative;

  display: inline-block;

  width: 20px;
  height: 20px;
  margin: 2px 8px 0 0;

  transform: scale(0.75);
  vertical-align: middle;

  color: var(--colors-darkBlue);
  border: 1px solid var(--colors-greekLilac);
  border-radius: 2px;
  background-color: inherit;
`;

export const StyledInput = styled.input`
  font-size: 1rem;
  line-height: 1.5;

  position: absolute;

  padding: 11px 23px;

  cursor: pointer;

  opacity: 0;
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 0;
  outline: 0;
  background-color: transparent;

  &:checked + ${StyledControl} {
    border-color: var(--colors-royalPurple);
  }

  &:checked + ${StyledControl}:after {
    position: absolute;
    top: 4px;
    left: 4px;

    display: block;

    width: 10px;
    height: 10px;

    content: '';

    border-radius: 2px;
    background-color: var(--colors-royalPurple);
  }
`;

export const StyledCheckbox = styled.label`
  font-weight: 500;

  display: flex;

  padding: 10px 12px;

  cursor: pointer;

  border: 1.5px solid var(--colors-greekLilac);
  border-radius: 12px;

  justify-content: center;
  align-items: center;
  align-items: flex-start;

  ${({ checked }) =>
    checked &&
    css`
      background-color: var(--colors-linkWater);
    `}

  &:hover {
    background-color: var(--colors-linkWater);
  }

  &:focus-within {
    box-shadow: 0px 0px 0px 2px var(--colors-irisBlue);
    background-color: var(--colors-linkWater);
  }
`;

export const StyledLabel = styled.span`
  font-weight: 500;
  font-size: 14px;

  color: var(--darkBlue);
`;
