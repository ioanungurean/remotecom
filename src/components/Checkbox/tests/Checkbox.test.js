import { render, screen, fireEvent } from '@testing-library/react';
import Checkbox from '../index';

describe('Checkbox', () => {
  it('renders correctly', () => {
    const { container } = render(<Checkbox id="checkbox" label="Checkbox" />);

    expect(container).toMatchSnapshot();
  });

  it('shows the correct label', () => {
    render(<Checkbox id="checkbox" label="Checkbox" />);

    expect(screen.getByText('Checkbox')).toBeInTheDocument();
  });

  it('calls the onChange handler', () => {
    const id = 'test';
    const spy = jest.fn();
    render(<Checkbox id={id} label="Checkbox" onChange={spy} />);
    fireEvent.click(screen.getByRole('checkbox'));
    expect(spy).toBeCalledWith(id);
  });
});
