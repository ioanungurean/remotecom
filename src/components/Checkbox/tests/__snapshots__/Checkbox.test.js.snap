// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div>
  <label
    class="sc-dlfnbm kyDBUR"
    for="checkbox"
  >
    <input
      class="sc-gsTCUz bYQViw"
      id="checkbox"
      type="checkbox"
    />
    <span
      class="sc-bdfBwQ dTBnQO"
    />
    <span
      class="sc-hKgILt hCmaEo"
    >
      Checkbox
    </span>
  </label>
</div>
`;
